store/store.go

```diff
@@ -41,40 +41,43 @@ type OpenOptions struct {
 }
 
 // OpenStore creates new FS Store under the given 'path'.
-// To be opened the Store must be initialized first, otherwise ErrNotInited is thrown.
+// To be opened, the Store must be initialized first, otherwise ErrNotInited is thrown.
 // OpenStore takes a file Lock on directory, hence only one Store can be opened at a time under the
 // given 'path', otherwise ErrOpened is thrown.
+// The store is locked only in the case of also opening the data store, however, in the case
+// of the keys, the store can still be opened.
 func OpenStore(logger tmlog.Logger, path string, options OpenOptions) (*Store, error) {
 	path, err := storePath(path)
 	if err != nil {
 		return nil, err
 	}
 
-	flock, err := fslock.Lock(lockPath(path))
-	if err != nil {
-		if err == fslock.ErrLocked {
-			return nil, ErrOpened
-		}
-		return nil, err
-	}
-
 	ok := IsInit(logger, path, InitOptions{
 		NeedDataStore:   options.HasDataStore,
 		NeedEVMKeyStore: options.HasEVMKeyStore,
 		NeedP2PKeyStore: options.HasP2PKeyStore,
 	})
 	if !ok {
-		flock.Unlock() //nolint: errcheck
 		return nil, ErrNotInited
 	}
 
 	var ds *badger.Datastore
+	var flock *fslock.Locker
 	if options.HasDataStore {
+		flock, err = fslock.Lock(lockPath(path))
+		if err != nil {
+			if err == fslock.ErrLocked {
+				return nil, ErrOpened
+			}
+			return nil, err
```

Comment:
I know this was just copied over, but do we need this extra logic just to change the error type?

Reply:
I didn't wanna change anything since i took it from the node, and it works as expected